### PR TITLE
[BugFix] Recover with empty tablet when BE is down

### LINKED ISSUE #5708: `recover_with_empty_tablet` dosen't work when BE is down with single replica
### Steps to reproduce the behavior (Required)
1. CREATE TABLE with single replica on cluster(1 FE, 2 BE)
```
CREATE TABLE lmh.`test` (
  `id` bigint(20),
  `name` varchar(65533),
  `dt` date NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`id`)
COMMENT "OLAP"
PARTITION BY RANGE(dt)(
   PARTITION p20220323 VALUES LESS THAN ("2022-03-24"),
   PARTITION p20220324 VALUES LESS THAN ("2022-03-25"),
   PARTITION p20220325 VALUES LESS THAN ("2022-03-26")
)
DISTRIBUTED BY HASH(`id`) BUCKETS 1
PROPERTIES (
  "replication_num" = "1"
);
```
2. INSERT data
```
insert into lmh.test VALUES(1, 'test', '2022-03-23'), (2, 'test', '2022-03-24'), (3, 'test', '2022-03-25');
```
3. kill any one BE
4. execute SELECT
```
select * from lmh.test;
```

### Expected behavior (Required)
`recover_with_empty_tablet` works, return remaining valid data

### Real behavior (Required)
```
mysql> select * from lmh.test;
ERROR 1064 (HY000): Backend not found. Check if any backend is down or not
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 ```
mysql> select current_version();
+-------------------+
| current_version() |
+-------------------+
| UNKNOWN bae30ec   |
+-------------------+
```

### PR DESCRIPTION
**Pull request by mchades**
## What type of PR is this：
- [x] bug

## Which issues of this PR fixes ：

Fixes #5708

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 27 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|0|23|00.00%|

**Comment by chaoyli** [comment]
What's the purpose of getting results from an empty tablet?

**Comment by mchades** [comment]
> What's the purpose of getting results from an empty tablet?

Just wanna the query can be executed normally even if there is data loss. @chaoyli

**Comment by chaoyli** [comment]
Config.recover_with_empty_tablet is used to recover a tablet that lost some data.
That makes the query run ok but the user should ingest the lost data again.
Your purpose is to make the query ignore the backends down when the replica has no data.
There are not the same. @mchades

**Comment by mchades** [comment]
> That makes the query run ok but the user should ingest the lost data again.

It's almost impossible to ingest the lost data again, because it's hard to tell the lost data.

When I create a single replica table and turn on Config.recover_with_empty_tablet, that means I can accept the cost of data loss but the query(select and insert) should not report error. However, at present, the table is disavailable until BE up. @chaoyli

**Comment by chaoyli** [comment]
@mchades There is some difference between BE is down and BE lost some data. When be is down, they should clone from other BEs. When BE lost some data, and can not be cloned from other BEs. You can use it.

**Comment by mchades** [comment]
> @mchades There is some difference between BE is down and BE lost some data. When be is down, they should clone from other BEs. When BE lost some data, and can not be cloned from other BEs. You can use it.

exactly correct! All works well with non-single replica. 

This PR is only considered about tablet with **single replica**:
> When be is down, they should clone from other BEs.

When BE is down, with **single replica**, there is no available source replica left, and resulte in the clone task fail. For user, the select and insert statement returns error on this tablet, it looks like `Config.recover_with_empty_tablet` dosen't work. It's the situation what u say as `When BE lost some data, and can not be cloned from other BEs`.
@chaoyli

**Comment by chaoyli** [comment]
I see it. Very glad to approve the pull request. Before it, you can add more FE unit tests to pass the test coverage. @mchades

**Review by chaoyli — DISMISSED** [review]

**Comment by mchades** [comment]
@chaoyli Thanks for your time to discussion. Coverage check has passed.

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/clone/TabletScheduler.java [hunk 1]
```diff
@@ -615,7 +615,8 @@ private void scheduleTablet(TabletSchedCtx tabletCtx, AgentBatchTask batchTask)
         }
     }
 
-    private void handleTabletByTypeAndStatus(TabletStatus status, TabletSchedCtx tabletCtx, AgentBatchTask batchTask)
+    // 'public' for test only
+    public void handleTabletByTypeAndStatus(TabletStatus status, TabletSchedCtx tabletCtx, AgentBatchTask batchTask)
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/clone/TabletScheduler.java:619]: You can use @VisibleForTesting annotation from guava to mark this situation.

Comment (mchades) [reply fe/fe-core/src/main/java/com/starrocks/clone/TabletScheduler.java:620]: done

**Review by Astralidea — DISMISSED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by chaoyli** [comment]
run starrocks_fe_unittest

**Comment by mchades** [comment]
run starrocks_fe_unittest

**Comment by mchades** [comment]
run starrocks_fe_unittest

**Comment by mchades** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 40 / 40 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|35|35|100.00%|

**Review by chaoyli — APPROVED** [review]

**Pull request merged**